### PR TITLE
Remove hard-coded git-commit from User-agent header

### PR DESCRIPTION
**Pull request by abaiken**
This will get the commit and pass it in as a build-arg when the image is built - which is then passed in as an ENV variable. We can then grab that in the ansible roles and set it in the User-agent header. I also added some helpful make commands and cleaned up the README to make more sense.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,7 +9,57 @@ endif
...
+	oc create -f deploy/crds/authentication_secret.yaml
+	oc create -f deploy/service_account.yaml
+	oc create -f deploy/role.yaml
+	oc create -f deploy/role_binding.yaml
+
+deploy-custom-resources:
+	oc create -f deploy/crds/cost_mgmt_cr.yaml
+	oc create -f deploy/crds/cost_mgmt_data_cr.yaml
+
+delete-dependencies-and-resources:
```
Comment (chambridge) [inline Makefile]: I don't think you ever clean up the operator metering report resources the operator creates. Might be nice to have  a target to do that.

Comment (abaiken) [reply Makefile]: That's a great idea! I will add one 😸

Comment (abaiken) [reply Makefile]: Added!

File: Makefile [hunk 2]
```diff
@@ -63,3 +64,24 @@ delete-dependencies-and-resources:
...
+	oc delete reportquery cm-openshift-persistentvolumeclaim
+	oc delete reportquery cm-openshift-persistentvolumeclaim-lookback
+	oc delete reportquery cm-openshift-usage
+	oc delete reportquery cm-openshift-usage-lookback
+	oc delete reportquery cm-persistentvolume-labels-raw
+	oc delete reportquery cm-persistentvolumeclaim-labels-raw
+	oc delete reportquery cm-pod-cpu-limit-raw
+	oc delete reportquery cm-pod-labels-raw
+	oc delete reportquery cm-pod-memory-limit-raw
+	oc delete reportquery cm-pod-persistentvolumeclaim-info-raw
```
Comment (chambridge) [inline Makefile]: Wouldn't it be simpler to just reference the directory so we don't have to keep up with a growing list?
```suggestion
	oc delete -f roles/setup/files
```

Comment (abaiken) [reply Makefile]: Yes! That is much better. Committed!

**Pull request merged**